src/trace_ng/log.zig

```diff
@@ -0,0 +1,672 @@
+const std = @import("std");
+const Level = @import("level.zig").Level;
+const logfmt = @import("logfmt.zig");
+const sig = @import("../sig.zig");
+// TODO Improve import.
+const Channel = @import("../sync/channel.zig").Channel;
+const testing = std.testing;
+const Allocator = std.mem.Allocator;
+const AtomicBool = std.atomic.Value(bool);
+const RecycleFBA = sig.utils.allocators.RecycleFBA;
+
+pub const Config = struct {
+    max_level: Level = Level.debug,
+    allocator: std.mem.Allocator,
+    /// Maximum memory that logger can use.
+    max_buffer: u64,
+    exit_sig: *std.atomic.Value(bool),
+    kind: LogKind = LogKind.standard,
+};
+
+const INITIAL_LOG_CHANNEL_SIZE: usize = 1024;
+
+const LogKind = enum {
+    standard,
+    testing,
+    noop,
+};
+
+const UnscopedLogger = StandardLogger(null);
+pub fn StandardLogger(comptime scope: ?type) type {
```

Comment:
consider changing the scope to a string.

Reply:
```zig
logger.withScope(@typeName(@This()));
```

doesn't seem too bad